src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -702,6 +709,42 @@ impl WindowState {
         })
     }
 
+    pub fn set_custom_cursor(&mut self, cursor: CustomCursor) {
+        let cursor = CustomCursorInternal::new(&mut self.pool, &cursor.inner);
+
+        if self.cursor_visible {
+            self.apply_custom_cursor(&cursor);
+        }
+
+        self.selected_cursor = SelectedCursor::Custom(cursor);
+    }
+
+    pub fn apply_custom_cursor(&self, cursor: &CustomCursorInternal) {
+        self.apply_on_poiner(|pointer, _| {
+            let surface = pointer.surface();
+
+            let scale = surface
+                .data::<SurfaceData>()
+                .unwrap()
+                .surface_data()
+                .scale_factor();
+            surface.set_buffer_scale(scale);
+            surface.attach(Some(cursor.buffer.wl_buffer()), 0, 0);
+            surface.damage_buffer(0, 0, cursor.w, cursor.h);
+            surface.commit();
+
+            let serial = pointer
+                .pointer()
+                .data::<WinitPointerData>()
+                .and_then(|data| data.pointer_data().latest_enter_serial())
+                .unwrap();
+
+            pointer
+                .pointer()
+                .set_cursor(serial, Some(surface), cursor.hotspot_x, cursor.hotspot_y);
```

Comment:
hotspot is surface local and not buffer coordinates, you need to convert.

Reply:
Fixed